nav2_costmap_2d/plugins/costmap_filters/costmap_filter.cpp

```
@@ -0,0 +1,119 @@
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the <ORGANIZATION> nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Alexey Merzlyakov
+ *********************************************************************/
+
+#include "nav2_costmap_2d/costmap_filters/costmap_filter.hpp"
+
+#include <exception>
+
+namespace nav2_costmap_2d
+{
+
+CostmapFilter::CostmapFilter()
+: is_rolling_(true), costmap_filter_info_topic_("/costmap_filter_info")
+{
+}
+
+CostmapFilter::~CostmapFilter()
+{
+}
+
+void CostmapFilter::onInitialize()
+{
+  // Get parameters
+  declareParameter("enabled", rclcpp::ParameterValue(true));
+  declareParameter("costmap_filter_info_topic");
+
+  node_->get_parameter(name_ + "." + "enabled", enabled_);
+  if (
```

If you just get via `value = get_parameter(...)` it will thrown an exception with full context, this is alot less context. Let it throw its own exception please

---

After adding the exception, I found information using only by it not enough. So, adding some error logging and re-throwing an incoming exception for best UX information: https://github.com/ros-planning/navigation2/pull/1882/commits/cfeeb1a0b96560bb68fa8f8119e27a62c65e7100#diff-9abde77dd5850f2da94347098f064bf5R62